### PR TITLE
refactor(cxx_indexer): DependentName nodes and edges

### PR DESCRIPTION
**Pull request by shahms**
BuildNodeIdForDependentName did that and all-the-things.  This results in hard-to-follow control flow and potentially redundant facts in the graph.  Pull this apart (mostly) into its constituent pieces.

### CODE REVIEW

File: kythe/cxx/indexer/cxx/IndexerASTHooks.cc [hunk 1]
```diff
@@ -1702,22 +1717,26 @@ bool IndexerASTVisitor::VisitDeclRefExpr(const clang::DeclRefExpr* DRE) {
 
 bool IndexerASTVisitor::VisitBuiltinTypeLoc(clang::BuiltinTypeLoc TL) {
   if (FLAGS_emit_anchors_on_builtins) {
-    return EmitTypeLocNodes(TL);
+    RecordTypeLocSpellingLocation(TL);
+    return true;
```
Comment (zrlk) [inline kythe/cxx/indexer/cxx/IndexerASTHooks.cc]:
```suggestion
```

Comment (shahms) [reply kythe/cxx/indexer/cxx/IndexerASTHooks.cc]: The linter doesn't care about PR titles, only commits.

Comment (shahms) [reply kythe/cxx/indexer/cxx/IndexerASTHooks.cc]: Also, the length requirements lead to less readable commit messages.

Comment (zrlk) [reply kythe/cxx/indexer/cxx/IndexerASTHooks.cc]: I'm not claiming that the linter is *good*, just that it's *angery*.

**Pull request merged**